### PR TITLE
signal.bsplines test #6835

### PR DESCRIPTION
**Pull request by cynthia-rempel**
Wrote a test-suite for scipy.signal.bsplines

Ran it from the scipy/scipy/signal/tests directory...

It yields a 98% coverage rating...
Used pylint to correct the formatting...
Made the changes indicated...

The test oracle right now are the results I got when running the functions in an interpreter... so they'll only catch changes in the code...

Really appreciate all you do!

### CODE REVIEW

File: scipy/signal/tests/test_bsplines.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+import numpy as np
+from numpy import array, nan
+from numpy.testing import (TestCase, assert_raises, run_module_suite,
+                           assert_allclose, assert_array_equal,
+                           assert_equal, assert_almost_equal)
+
+import scipy.signal.bsplines as bsp
+
+class TestBSplines(TestCase):
```
Comment (Eric89GXL) [inline scipy/signal/tests/test_bsplines.py]: Two newlines before class definition. Can you run `flake8` on this file and see if there are any other violations? Not all SciPy files conform but we try to make new additions abide by standard style if possible.

File: scipy/signal/tests/test_bsplines.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+    def test_spline_filter(self):
+        np.random.seed(12457)
+        # Test the type-error branch
+        assert_raises(TypeError, bsp.spline_filter, array([0]), 0)
+        # Test the complex branch
+        data_array_complex = np.random.rand(7, 7) + np.random.rand(7, 7)*1j
+        # make the magnitude exceed 1, and make some negative
+        data_array_complex = 10*(1+1j-2*data_array_complex)
+        result_array_complex = array(
```
Comment (Eric89GXL) [inline scipy/signal/tests/test_bsplines.py]: Can you add a comment about where these veridical values come from (MATLAB, other package, etc.)

File: scipy/signal/tests/test_bsplines.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+              -7.71144306, 3.78263161, 6.46034818, -4.43444731]])
+        assert_allclose(bsp.spline_filter(data_array_real, 0), result_array_real)
+
+    def test_bspline(self):
+        np.random.seed(12458)
+        assert_allclose(bsp.bspline(np.random.rand(1, 1), 2), array([[0.73694695]]))
+        data_array_complex = np.random.rand(4, 4) + np.random.rand(4, 4)*1j
+        # scaling the magnitude by 10 makes the results close enough to zero,
+        # that the assertion fails, so just make the elements have a mix of
+        # positive and negative imaginary components...
```
Comment (Eric89GXL) [inline scipy/signal/tests/test_bsplines.py]: You shouldn't need to play with scale here. This is why `assert_allclose` has `rtol` and `atol` arguments

**Review by larsoner — COMMENTED** [review]
Otherwise LGTM

**Comment by ev-br** [comment]
https://github.com/scipy/scipy/issues/6393 is likely related (or incorrect)

File: scipy/signal/tests/test_bsplines.py (hunk 2 above)
Comment (vyasr) [reply scipy/signal/tests/test_bsplines.py]: @rgommers I can fix the other parts of this PR but I don't know how she set this up initially. This is true about all the values that are being used for verification.

Comment (rgommers) [reply scipy/signal/tests/test_bsplines.py]: I suspect that @cynthia-rempel just took the output of the current `scipy.signal` functions here. Would be great if she could confirm, but let's not wait for that. If the tests pass I'm okay with adding them as pure regression tests and a comment to that effect (e.g. "testing against currently returned values as of scipy 1.1.0, for regression purposes").

**Comment by vyasr** [comment]
These changes have been incorporated into pull request #9035.

**Comment by vyasr** [comment]
@rgommers this PR can be closed now.

**Comment by rgommers** [comment]
Thanks @vyasr. And thanks @cynthia-rempel as well!

**Pull request closed without merging**